File: Collisions/2211. Count Collisions on a Road/2211. Count Collisions on a Road.py
class Solution:
    def countCollisions(self, directions: str) -> int:
        rets = 0
        n = len(directions)
        flag = 0;
        for i in range(n):
            if flag ==0 and (directions[i] == 'R' or directions[i] == 'S'):
                flag = 1
            if flag == 1 and directions[i] == 'L':
                rets += 1

        flag = 0
        for i in range(n - 1, -1, -1):
            if flag ==0 and (directions[i] == 'L' or directions[i] == 'S'):
                flag = 1
            if flag == 1 and directions[i] == 'R':
                rets += 1
        
        return rets

File: Collisions/2211. Count Collisions on a Road/2211. Count Collisions on a Road.cpp
class Solution {
public:
    int countCollisions(string directions) {
        int rets = 0;
        int n = directions.size();
        int flag = 0;
        for (int i = 0; i < n; i++) {
            char c = directions[i];
            if (flag ==0 && (c == 'R' || c == 'S')) {
                flag = 1;
            }
            if (flag == 1 && c == 'L') {
                rets++;
            }
        }

        flag = 0;
        for (int i = n-1; i >= 0; i--) {
            char c = directions[i];
            if (flag ==0 && (c == 'L' || c == 'S')) {
                flag = 1;
            }
            if (flag == 1 && c == 'R') {
                rets++;
            }
        }
        return rets;
    }
};
